### PR TITLE
Feature/new test strategy

### PR DESCRIPTION
**Pull request by kem198**
This pull request focuses on enhancing the `TodoRestControllerTests` class by introducing comprehensive test cases for various API endpoints and functionalities. Additionally, it removes outdated or redundant test classes (`FizzBuzzServiceTests`, `GreetingServiceTests`, and `TodoServiceTests`) that are no longer relevant to the current implementation. 

### Enhancements to `TodoRestControllerTests`:

* **Addition of new test cases for CRUD operations:**
  - Tests for retrieving all todos (`GetTodosTests`) and specific todos by ID (`GetTodoTests`) with proper assertions for HTTP status codes and response content.
  - Tests for creating todos (`PostTodoTests`), including validation for database persistence, handling unfinished todo limits, and validation errors.
  - Tests for marking todos as finished (`PutTodoFinishTests`), ensuring correct database updates and handling edge cases like already finished todos.
  - Tests for deleting todos (`DeleteTodoTests`) with validations for successful deletion and handling non-existent todo cases.

* **Integration of `TodoRepository`:**
  - Added `TodoRepository` as a dependency to facilitate direct database interactions during testing. This ensures tests cover both API behavior and database state changes. [[1]](diffhunk://#diff-f3ff1055062c368e420d9552eec39430e9717d159021a68ac3f29a76023c005fR25) [[2]](diffhunk://#diff-f3ff1055062c368e420d9552eec39430e9717d159021a68ac3f29a76023c005fR35-R127)

* **Helper method for creating todos:**
  - Introduced a reusable `createTodo` method to streamline test setup and reduce code duplication.

### Removal of redundant test classes:

* **Deleted `FizzBuzzServiceTests`:**
  - Removed tests for the FizzBuzz functionality, which are no longer relevant to the current scope of the project.

* **Deleted `GreetingServiceTests`:**
  - Removed tests for the Greeting service, as this functionality has been deprecated or moved out of the current project context.

* **Deleted `TodoServiceTests`:**
  - Removed service-level tests for `TodoService`, as the focus has shifted to testing through the controller and repository layers.

### CODE REVIEW

File: src/todos-api/src/test/java/net/kem198/todos_api/api/todo/TodoRestControllerTests.java [hunk 1]
```diff
@@ -28,16 +32,99 @@ public class TodoRestControllerTests {
...
     private TestRestTemplate restTemplate;
 
+    @Autowired
+    private TodoRepository todoRepository;
+
     @BeforeEach
     void setUp() {
         objectMapper = new ObjectMapper();
+        // テストデータをクリア
+        todoRepository.findAll().forEach(todo -> todoRepository.delete(todo));
```
[the remark this answers is not in the document]
Comment (kem198) [reply src/todos-api/src/test/java/net/kem198/todos_api/api/todo/TodoRestControllerTests.java:44]: そのメソッドは作ってないよ
